### PR TITLE
Update typescript to 3.0.1

### PR DESCRIPTION
**Pull request by antonva**
TypeDoc is updated to 0.12.0 as that's the latest release to support TS 3.0.

### CODE REVIEW

File: src/excmds.ts [hunk 1]
```diff
@@ -1763,17 +1763,24 @@ export async function tabclosealltoleft() {
...
-    const sessions = await browser.sessions.getRecentlyClosed()
+    const sessions = await browser.sessions.getRecentlyClosed({ maxResults: 10 })
 
     // The first session object that's a window or a tab from this window. Or undefined if sessions is empty.
-    let closed = sessions.find(s => {
-        return "window" in s || (s.tab && s.tab.windowId == current_win_id)
+    const lastSession = sessions.find(s => {
+        if (s.window) {
+            return true
+        } else if (s.tab && s.tab.windowId) {
```
Comment (bovine3dom) [inline src/excmds.ts]: Why did you move check down? If I close tabs in another window I don't want undo not to work...

Comment (antonva) [reply src/excmds.ts]: Oh right, oversight on my part. Got interrupted while rewriting that. The check was moved down in another implementation where it would have made sense.

**Comment by bovine3dom** [comment]
LGTM :)

**Comment by bovine3dom** [comment]
NPM complains about

`npm WARN ts-jest@21.1.3 requires a peer of typescript@2.x but none is installed. You must install peer dependencies yourself.`

should I be worried?
